utils.go

```diff
@@ -47,8 +47,10 @@ func getAuth(req *http.Request) (string, string) {
 //
 // getQuerySnippet must be called only for error reporting.
 func getQuerySnippet(req *http.Request) string {
-	if req.Method == http.MethodGet {
-		return req.URL.Query().Get("query")
+	var query string
+
+	if req.URL.Query().Get("query") != "" {
+		query = req.URL.Query().Get("query")
 	}
 
 	crc, ok := req.Body.(*cachedReadCloser)
```

Comment:
you must check if request is POST or body is not nil. Otherwise process will panic

Reply:
I revised the source code of HTTP lib. There is a comment that it should never happen. You can check it yourself
https://golang.org/src/net/http/request.go?s=3211:11861#L174